src/main/java/io/reactivex/NbpObserver.java

```diff
@@ -0,0 +1,41 @@
+/**
+ * Copyright 2015 Netflix, Inc.
+ * 
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in
+ * compliance with the License. You may obtain a copy of the License at
+ * 
+ * http://www.apache.org/licenses/LICENSE-2.0
+ * 
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is
+ * distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See
+ * the License for the specific language governing permissions and limitations under the License.
+ */
+
+package io.reactivex;
+
+import io.reactivex.NbpObservable.NbpSubscriber;
+import io.reactivex.disposables.Disposable;
+import io.reactivex.internal.subscriptions.SubscriptionHelper;
+
+public abstract class NbpObserver<T> implements NbpSubscriber<T> {
+    private Disposable s;
```

Comment:
No. onSubscribe must be called sequentially, similar to the full-blown RS Subscriber. There is an NbpAsyncObserver where this is indeed volatile because the cancellation is exposed and may happen before a Disposable is set.

If merged, you can run around and rename stuff more easily because of tools. When I type in such classes, it is such a burden to type out everything.